framework/data/Sort.php

```diff
@@ -243,8 +243,8 @@ public function getAttributeOrders($recalculate = false)
                 $request = Yii::$app->getRequest();
                 $params = $request instanceof Request ? $request->getQueryParams() : [];
             }
-            if (isset($params[$this->sortParam]) && is_scalar($params[$this->sortParam])) {
-                $attributes = explode($this->separator, $params[$this->sortParam]);
+            if (isset($params[$this->sortParam])) {
+                $attributes = $this->parseSortAttributes($params[$this->sortParam]);
```

Comment:
`parseSortParam` would be a better name, because that is what it does, it "parses the sort param" and returns the attributes.